kernel/tests/read.rs

```diff
@@ -120,7 +120,10 @@ async fn single_commit_two_add_files() -> Result<(), Box<dyn std::error::Error>>
     let scan = snapshot.into_scan_builder().build()?;
 
     let mut files = 0;
-    let stream = scan.execute(&engine)?.into_iter().zip(expected_data);
+    let stream = scan
```

Comment:
these tests return a result which gives a slightly nicer message on failure. So I think it's worth the bit of extra code to do:
```rust
    for res in stream {
        let (data, expected) = res?; 
        let raw_data = data.raw_data?;
        ...
```

and not map the unwrap.  same below.